modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/BusyExecutor.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.query.stat;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.internal.util.GridBusyLock;
+import org.apache.ignite.thread.IgniteThreadPoolExecutor;
+
+/**
+ * Executor with busy run support.
+ * Can run any tasks while active and safelly wait untill they stopped.
+ */
+public class BusyExecutor {
```

Comment:
What reason to have the class? I'm sure that implicit lock at concrete place will be more understandable and safe. Right now we could lock all works in case problem in one place (thread  take a lock and dead, as example). Or how it will work in multi JVM environment?
Let's move locks to place where it uses.

Reply:
1) Reason to create this class was simplicity - to move taking busy lock and checking working status to some wrapper class.
2) Lock all works in case of problem in one place - is just how our's BusyLock or traditional ReentrantReadWriteLock works. And no matter, is it used from the same class or from separated BusyExecutor. Moreover, BusyExecuter can guarantee that all task will be wrapped in try/catch block with leaveBusy() at final section.
3) It definitely won't work in multi JVM environment because it shouldn't - we have no multiJVM nodes.
Are you sure we need to move BusyLock back to StatisticsManager, GlobalStatisticsManager, StatisticsProcessor?